logstash-core/lib/logstash/environment.rb

```diff
@@ -15,7 +15,7 @@ module Environment
     end
   end
 
-  [
+  LOGSTASH_SETTINGS = [
```

Comment:
I may be missing something but I don't see the need for this refactor

The `register_settings` method is only used in tests and right here, and even in the tests the support_escapes setting isn't done through a Settings object, but directly with the setter https://github.com/elastic/logstash/pull/7442/files#diff-1c97814b03bd1f3c0f796071ebc0cf19R169

Maybe this refactor isn't necessary?